### ISSUE TITLE
Check links on click

### ISSUE THREAD

**Issue by Jack28** [SB0599/Phishing-E-Mail-Checker#1]
Hi, this is only a thought and maybe a feature request.

Recently I thought about redirect URLs and if it would be possible to check them. Perhaps similar to shortened URLs.
The problem also seems to be that a URL can not be requested (with neither HTTP METHOD) without risking to trigger a possibly unwanted action.

Then I thought of untrackme for android (https://f-droid.org/packages/app.fedilab.nitterizeme/) that automatically redirects links to websites to their libre alternative.

Would it be possible to register Phishing-E-Mail-Checker for any URL and quickly enough check them when they are clicked?

Common problems are avoided, so no edits on the message, breaking of signatures, even encryption wouldn't be a problem.

I'm curious to read your opinion.

Best regards

**Comment by SB0599** [comment]
As the Phishing-E-Mail-Checker does not act as an e-mail client, it does not make sense to implement this function in it. It is purely aimed at scanning and searching the E-Mail. This means that no URL can be checked when it is clicked on in any E-Mail-Client. The URLs are already checked when the message is received. 
The problem of redirecting can be solved in this checker, for example, by implementing a URL check that calls the URL and watch where the request is redirected to. 
Of course, this cannot avoid the problem of encryption and brings further problems with it.

One idea would be another function in the app or an extra application that creates a VPN on the phone which can check the requests in real time (comparison to the DuckDuckGo app tracker protection).

I hope I could help with my answer.  If you have any further questions, just write me

Best regards

**Comment by Jack28** [comment]
I agree, it is not within the scope of Phishing-E-Mail-Checker to provide this functionality.

However Phishing-E-Mail-Checker is an e-mail client!!
It just doesn't display messages or makes them accessible to the user in any way.

I like the idea of a separate app that would register for the appropriate intents and quickly check it before redirecting the intent to the actual application making use of the entire scan logic already present in this project.
If the purpose of Phishing-E-Mail-Checker is to only secure the mobile device the email functionality could maybe be substituted completely since images, links, and any sort of attachment are opened by an intent
(that likely doesn't include in-app previews and webviews but would certainly account for most mail clients and hence widen the protection significantly also covering other applications).
